base/timing.jl

```diff
@@ -86,6 +86,64 @@ end
 # total time spend in garbage collection, in nanoseconds
 gc_time_ns() = ccall(:jl_gc_total_hrtime, UInt64, ())
 
+"""
+    task_cpu_time_ns(t::Task) -> UInt64
+
+Return the total nanoseconds that the task `t` has spent running.
+This metric is only updated when the task yields or completes.
+See also [`task_wall_time_ns`](@ref).
+
+Will be `UInt64(0)` if task timings are not enabled.
```

Comment:
or `missing` or `nothing` are also possible options for that case

Reply:
IMO you should not use unsigned integers for representing positive values - you should use `Int64` for this instead.

It's better in julia than it is in C, since julia's defaults are less dangerous, but using unsigned ints for positive numeric values is a consistent source of bugs in C programs, and i think should be still frowned upon in julia. (https://jacobegner.blogspot.com/2019/11/unsigned-integers-are-dangerous.html)

IMO Unsigned ints should be used only for things like hashes, identifiers, addresses, and possibly for indexes but even for indexes i'd use signed ints.

But anyway, another benefit is that this also gives `-1` as a possible return value for the unset case.

Reply:
I've so far opted for returning an `Int` when enabled and `nothing` otherwise. I think `nothing` is maybe nicer because it'll force an error if you try to use it like an `Int`.

That said, these functions currently have` _ns()` names, similar to `cumulative_compile_time_ns()` and `gc_time_ns()` and `time_ns()` which all return `UInt64`, so don't love the inconsistency

I also moved these all to `Experimental` so we can change these APIs as we learn more about what makes sense (as we add more metrics).